storage/remote/write_handler.go

```diff
@@ -115,6 +120,12 @@ func (h *writeHandler) write(ctx context.Context, req *prompb.WriteRequest) (err
 
 	b := labels.NewScratchBuilder(0)
 	var exemplarErr error
+
+	timeLimitApp := &timeLimitAppender{
+		Appender: app,
```

Comment:
nitty nit: we can initialize the appender directly here and use timeLimitApp everywhere.